components/raftstore/src/store/peer.rs

```diff
@@ -68,35 +66,54 @@ pub enum StaleState {
 }
 
 /// Meta information about proposals.
-pub struct ProposalMeta {
-    pub index: u64,
-    pub term: u64,
+pub struct ProposalMeta<E>
+where
+    E: KvEngine,
+{
+    pub proposal: Proposal<E>,
```

Comment:
Instead of adding `Proposal` to `ProposalMeta` how about moving `renew_lease_time` to `Proposal`? So we can remove `ProposalMeta` completely.